fictionary/__init__.py

```diff
@@ -15,29 +15,36 @@
 import random
 import shelve
 import sys
+import tempfile
 
-import click
-
-APP_NAME = "fictionary"
 
 # Where to save the generated data file:
-DATA_FILE_ROOT = click.get_app_dir(APP_NAME)
+DATA_FILE_ROOT = tempfile.gettempdir()
 
 # Where to load the source ispell wordlists:
 SRC_DATA_FILE_ROOT = dirname(__file__)
 
-LOG = logging.getLogger(__name__)
+LOG = logging.getLogger('fictionary')
+
+DICT_ALL_KEY = str('all')
```

Comment:
What's the reason for wrapping all these in `str`? It won't provide database compatibility between 2 and 3, because in Python 2, str is bytes, and in Python 3, str is unicode.

In fact, now I think about it, that `str` wrapper will make no difference to the program at all. `'all'` is already of type `str`

Reply:
In general, though I like that you've extracted the database keys into constants!

Reply:
The reason behind all the `str()` wrapping, is because, as it turns out, it seems that `shelve` in Python 2.7 doesn't support Unicode shelf keys. And because we're using `from __future__ import unicode_literals`, all the literal shelf keys are effectively of the form `u'key-name'`. This was causing me errors (unfortunately I need to use Fictionary with an app that's still stuck on Python 2.7). By wrapping the keys in `str()`, they become non-Unicode strings in Python 2, and they remain as Unicode in Python 3. Normally you'd be right, normally the `str()` wrapper would make no difference - but because of the use of `unicode_literals` in Py2, actually it does change the behaviour.

And yes, you're right, it doesn't provide database compatibility - I can confirm that, when switching between Python 2 and 3, the database file needs to be deleted and re-created. But I'd say that this is to be expected.

Reply:
Have now made it so that the dictionary database has a different filename, when running in Python 2 vs Python 3. So if someone uses Fictionary on the same machine, with some Python 2 scripts and some Python 3 scripts, there should be no conflict / incompatibility.

Reply:
That's great!